integration/tests/cook/test_basic.py

```diff
@@ -1879,6 +1879,67 @@ def test_basic_docker_job(self):
         job = util.wait_for_job(self.cook_url, job_uuid, 'completed')
         self.assertIn('success', [i['status'] for i in job['instances']])
 
+    def test_submit_job_with_disk(self):
+        settings_dict = util.settings(self.cook_url)
+        valid_disk_types_config_map = settings_dict.get("pools", {}).get("valid-disk-types", [])
+        # If no pools support disks, submit a job to the default pool and assert the submission gets rejected
+        if not valid_disk_types_config_map:
+            default_pool = util.default_submit_pool()
+            job_uuid, resp = util.submit_job(
+                self.cook_url,
+                pool=default_pool,
+                disk={'size': 20000, 'type': 'pd-ssd'})
+            self.assertEqual(resp.status_code, 400)
+            self.assertTrue(f"The following disk type is not supported: pd-ssd" in resp.text,
+                msg=resp.content)
+        else:
+            # Check if there are any active pools
+            active_pools, _ = util.active_pools(self.cook_url)
```

Comment:
Not worth considering if active_pools is empty. If thats true either we're configured for no pools (which is not production) or we're completely busted, in which case not interesting.

Reply:
interesting. for gpu testing, we followed a similar structure of checking if there are any active pools. is that case also useless then?

Reply:
Also interesting. I guess its fine, just seems strange and we should explain why...
Ah hah do we need to add something like this there (and here)?:
"""
GPU's may not be enabled for COOK_TEST_DEFAULT_SUBMIT_POOL, so we iterate over all active pools to find a pool where we can run this test.
"""

Reply:
I think it's worth adding to the gpu test but dpo and I think it is probably overkill for the disk test

Reply:
I meant that as a proposed comment to add here (and at the GPU test). If thats not the reason why we're iterating, than can you put in a comment explaining why?

Reply:
ah, misunderstood your comment. added in the explanation as a comment